sdk/lib/rtl/heap.c

```diff
@@ -280,39 +269,92 @@ RtlpInsertFreeBlockHelper(PHEAP Heap,
         FreeEntry->Flags &= HEAP_ENTRY_LAST_ENTRY;
     }
 
-    /* Insert it either into dedicated or non-dedicated list */
-    if (BlockSize < HEAP_FREELISTS)
+    /* See if this should go to the dedicated list */
```

Comment:
This code (everytime FreeHints is used) feels insecure when `DeCommitFreeBlockThreshold` happens to be equal to zero.